utils/biospecimen.js

```diff
@@ -649,10 +659,84 @@ const biospecimenAPIs = async (req, res) => {
+        const { assignKitToParticipant } = require('./firestore');
+        const response = await assignKitToParticipant(requestData);
+        if(!response) {
+            return res.status(404).json({data: response, code:404});
+        } else if (response instanceof Error) {
+           return res.status(500).json({data: response, code: 500});
+        } else {
+            return res.status(200).json({data: response, code:200});
+        }
+    }
+
+    else if(api == 'verifyScannedCode'){
+        if(req.method !== 'GET') {
+            return res.status(405).json(getResponseJSON('Only GET requests are accepted!', 405));
+        }
+        const query = req.query.id
+        if(Object.keys(query).length === 0) return res.status(404).json(getResponseJSON('Please include id to verify scanned code.', 400));
+        const { processVerifyScannedCode } = require('./firestore');
+        const response = await processVerifyScannedCode(query);
+        if(!response) {
+            return res.status(404).json({data: response, code:404});
+        } else if (response instanceof Error) {
+           return res.status(500).json({data: response, code: 500});
+        } else {
+            return res.status(200).json({data: response, code:200});
+        }
     }
 
+    else if(api === 'confirmShippment'){
+        if(req.method !== 'POST') {
+            return res.status(405).json(getResponseJSON('Only POST requests are accepted!', 405));
+        }
+        let requestData = req.body;
+        if(Object.keys(requestData).length === 0 ) return res.status(400).json(getResponseJSON('Request body is empty!', 400));
+        const { confirmShippmentKit } = require('./firestore');
+        const response = await confirmShippmentKit(requestData);
+        if(!response) {
+            return res.status(404).json({data: response, code:404});
+        } else if (response instanceof Error) {
+            return res.status(500).json({data: response, code: 500});
+        } else {
+            return res.status(200).json({message: `Success!`, code:200})
+        }
+    }
+
+    else if(api === 'kitReceipt') {
+        if(req.method !== 'POST') {
+            return res.status(405).json(getResponseJSON('Only POST requests are accepted!', 405));
+        }
+        let requestData = req.body;
+        if(Object.keys(requestData).length === 0 ) return res.status(400).json(getResponseJSON('Request body is empty!', 400));
+        const { storeKitReceipt } = require('./firestore');
+        const response = await storeKitReceipt(requestData);
+        if(!response) {
+            return res.status(404).json({data: response, code:404});
+        } else if (response instanceof Error) {
+            return res.status(500).json({data: response, code: 500});
+        } else {
+            return res.status(200).json({message: `Success!`, code:200})
+        }
```

Comment:
Please consider the same as above for structuring the error handling flow in the endpoint and the function it calls.